lib/temporal/worker.rb

```diff
@@ -0,0 +1,72 @@
+require 'async'
+require 'temporal/bridge'
+require 'temporal/data_converter'
+require 'temporal/failure_converter'
+require 'temporal/payload_converter'
+require 'temporal/runtime'
+require 'temporal/worker/activity'
+require 'temporal/worker/thread_pool_executor'
+
+module Temporal
+  class Worker
+    # TODO: Add worker interceptors
+    def initialize(
+      connection,
+      namespace,
+      task_queue,
+      payload_converter: Temporal::PayloadConverter::DEFAULT,
+      payload_codecs: [],
+      failure_converter: Temporal::FailureConverter::DEFAULT,
+      activity_executor: nil,
+      max_concurrent_activities: 100
+    )
+      @running = false
+      @mutex = Mutex.new
+      @runtime = Temporal::Runtime.instance
+      activity_executor ||= ThreadPoolExecutor.new(max_concurrent_activities)
+      data_converter = DataConverter.new(
+        payload_converter: payload_converter,
+        payload_codecs: payload_codecs,
+        failure_converter: failure_converter,
+      )
+      core_worker = Temporal::Bridge::Worker.create(
```

Comment:
Can you confirm w/ your Ruby lib that this is freed on the Rust when the worker is GC'd by Ruby?

Reply:
Is this effectively connecting (calling getSystemInfo) and blocking the calling thread?
Make sure to document this and consider later (much later) adding a static method to instantiate multiple threads.
In TS creating a worker throws if it can't establish a connection with the server.

Reply:
Right now we have an eager connection, so as far as gRPC goes we should already be connected by this point (via connection). Then we just call Core's `temporal_sdk_core::init_worker`, which I believe is blocking. We can switch to using a non-blocking version (from the reactor's perspective) later if that is a concern, but again from Ruby's perspective — it's better to have things blocking the flow, rather than force a callback style programming style